app/models/miq_worker/container_common.rb

```diff
@@ -44,15 +44,20 @@ def default_image
     def resource_constraints
       mem_threshold = self.class.worker_settings[:memory_threshold]
       cpu_threshold = self.class.worker_settings[:cpu_threshold_percent]
+      mem_default   = self.class.worker_settings[:memory_minimum]
+      cpu_default   = self.class.worker_settings[:cpu_minimum_percent]
 
-      return {} if !Settings.server.worker_monitor.enforce_resource_constraints || (mem_threshold.nil? && cpu_threshold.nil?)
+      return {} unless Settings.server.worker_monitor.enforce_resource_constraints
+      return {} unless mem_threshold || cpu_threshold || mem_default || cpu_default
 
-      {:limits => {}}.tap do |h|
-        h[:limits][:memory] = "#{mem_threshold / 1.megabyte}Mi" if mem_threshold
-        if cpu_threshold
-          millicores = ((cpu_threshold / 100.0) * 1000).to_i
-          h[:limits][:cpu] = "#{millicores}m"
-        end
+      {}.tap do |h|
+        h[:limits] = {} if mem_threshold || cpu_threshold
```

Comment:
To avoid this line you can just change the other lines to, for example...

```diff
- h[:limits][:memory] = format_memory_threshold(mem_threshold) if mem_threshold
+ h.store_path(:limits, :memory, format_memory_threshold(mem_threshold)) if mem_threshold
```